content/posts/deploying-application-on-k8s-with-helm/index.md

```diff
@@ -0,0 +1,431 @@
++++
+date = "2021-12-15T14:28:23+07:00"
+author = "sanglh"
+description = "How to work with helm chart and something about helm subchart/dependency feature"
+title = "Deploying application on Kubenetes with Helm"
+categories = ["DevSecOps", "Helm", "K8S"]
+tags = ["k8s", "helm", "kubernetes"]
++++
+
+## Why do we need to use helm chart?
+
+Working with k8s, we have many k8s resources: ConfigMap, Secret, Service, Deployment, Pod,... K8s resources can be created by creating the yaml file, and use command kubectl apply -f <link_to_the_file>. For example, we create a file with name is deployment.yaml:
+
+```yaml
+
+  apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: nginx-deployment
+  spec:
+    replicas: 2
+    selector:
+      matchLabels:
+        app: nginx-deployment
+    template:
+      metadata:
+        labels:
+          app: nginx-deployment
+      spec:
+        containers:
+        - name: nginx-deployment
+          image: nginx
+          ports:
+          - containerPort: 8080
+
+```
+To create the k8s deployment, we use the command:
+```bash
+
+  kubectl apply -f deployment.yaml
+
+```
+
+With each resource, we can create a file (deployment.yaml, service.yaml, configmap.yaml, secret.yaml,...) and use the kubectl command to apply them. But we have a problem. If we work with microservices and we have multiple services => the number of yaml files is very large, and it make us difficult to manage them.
+
+We need a solution to manage all of them in a common template => Helm can solve this problem
+
+Before go to next part, you need to download and install minikube on your local device.
+
+## Introduction to helm chart
+**Helm is a tool that streamlines installing and managing Kubernetes applications.** To work with helm chart, we need to install helm command. This link is about how to install: <a>https://helm.sh/docs/intro/install</a>
+
+I will introduce you to the structure of the helm chart. To create a chart, we can use my example command:
+```bash
+
+  helm create mynginx
+
+```
+That command will create an mynginx folder. Let take a look in mynginx folder, we have some folders and files:
+- charts: this folder will contain another subchart which our chart will depend on. I will talk more about subchart it in the next section
```

Comment:
talk more about that subchart in the next section / talk more about that in the next section